### PR TITLE
Amazon AWS Delete Subnet

### PR DESCRIPTION
**Pull request by mihaitusa**
Added Delete Subnet operation.
Signed off by Tusa Mihai: tusa.mihai@gmail.com

### CODE REVIEW

**Comment by mihaitusa** [comment]
@Bonczidai Please review it!

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                                       @Param(value = PROXY_HOST) String proxyHost,
+                                       @Param(value = PROXY_PORT) String proxyPort,
+                                       @Param(value = PROXY_USERNAME) String proxyUsername,
+                                       @Param(value = PROXY_PASSWORD, encrypted = true) String proxyPassword,
+                                       @Param(value = HEADERS) String headers,
+                                       @Param(value = QUERY_PARAMS) String queryParams,
+                                       @Param(value = VERSION) String version,
+                                       @Param(value = SUBNET_ID) String subnetId) {
+        try {
+            version = InputsUtil.getDefaultStringInput(version, "2016-09-15");
```
Comment (Bonczidai) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java]: same comment here for const

**Comment by Bonczidai** [comment]
great @mihaitusa - guess who's the man

let's wait for #233 since this PR is dependent on it

@Tirla-Alin @tethryus please also take a look on this PR

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
+package io.cloudslang.content.amazon.actions.network;
+
+import com.hp.oo.sdk.content.annotations.Action;
+import com.hp.oo.sdk.content.annotations.Output;
+import com.hp.oo.sdk.content.annotations.Param;
+import com.hp.oo.sdk.content.annotations.Response;
+import com.hp.oo.sdk.content.plugin.ActionMetadata.MatchType;
+import com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType;
+import io.cloudslang.content.amazon.entities.constants.Outputs;
```
Comment (Tirla-Alin) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java]: You should use CloudSlang commons.

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                                       @Param(value = IDENTITY, required = true) String identity,
+                                       @Param(value = CREDENTIAL, required = true, encrypted = true) String credential,
+                                       @Param(value = PROXY_HOST) String proxyHost,
+                                       @Param(value = PROXY_PORT) String proxyPort,
+                                       @Param(value = PROXY_USERNAME) String proxyUsername,
+                                       @Param(value = PROXY_PASSWORD, encrypted = true) String proxyPassword,
+                                       @Param(value = HEADERS) String headers,
+                                       @Param(value = QUERY_PARAMS) String queryParams,
+                                       @Param(value = VERSION) String version,
+                                       @Param(value = SUBNET_ID) String subnetId) {
```
Comment (Tirla-Alin) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java]: Isn't this required?

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java (hunk 1 above)
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java]: @Bonczidai Yeap... wait to see what @Tirla-Alin have to say about it.

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java (hunk 2 above)
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java]: @Tirla-Alin CloudSlang commons use I will....

File: cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java (hunk 3 above)
Comment (mihaitusa) [inline cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/network/DeleteSubnetAction.java]: @Tirla-Alin yeap... good catch!

**Comment by Tirla-Alin** [comment]
👍 Great job. Merging since all comments were handled.

**Pull request merged**